File: content/realm/examples/dart/test/person.dart
import 'package:realm_dart/realm.dart';

// :snippet-start: ejson-annotation
class Person {
  final String name;
  final DateTime birthDate;

  final int? age;
  final double income;
  final Person? spouse;

  @ejson // annotate constructor to generate decoder and encoder
  Person(this.name, this.birthDate, this.income, {this.spouse, this.age});
}
// :snippet-end:

File: content/realm/examples/react-native/legacy/__tests__/js/Models/Person.js
import Realm from 'realm';

// :snippet-start: js-person-schema
class Person extends Realm.Object {
  static schema = {
    name: 'Person',
    properties: {
      name: 'string',
      age: 'int?',
    },
  };
}
// :snippet-end:
export default Person;
